0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -52,7 +61,48 @@ Content-Type: application/octet-stream
 < Body: Binary OER-Encoded ILP Fulfill or Reject Packet >
 ```
 
-All ILP Packets MUST be returned with the HTTP status code `200: OK`.
+### ILP Fulfill/Reject
+
+#### Request
+
+```http
+POST /ilp HTTP/x.x
+Host: example.com
+Content-Type: application/octet-stream
+Authorization: Bearer zxcljvoizuu09wqqpowipoalksdflksjdgxclvkjl0s909asdf
+Idempotency-Key: 8988dd17-55e4-40e0-9c57-419d81a0e3a5
+
+< Body: Binary OER-Encoded ILP Fulfill or Reject Packet >
+```
+
+- **Path** &mdash; A connector MAY specify any HTTP path for their peer to send ILP packets to.
+- **Host Header** &mdash; The standard [HTTP Host Header](https://tools.ietf.org/html/rfc2616#section-14.23) indicating the domain of the HTTP server the Request is sent to.
+- **Content-Type Header** &mdash; MUST be set to `application/octet-stream`.
+- **Idempotency Key Header** &mdash; Same idempotency key used in the corresponding ILP Prepare.
```

Comment:
I'm not an expert on the use of idempotency keys, but this seems like it's overloading the same ID with both the idempotency behavior and a way to correlate a given request/response.

Is it better to:
- Have separate IDs for idempotency keys and request/response correlation
- Have one ID and call it the `Idempotency-Key`
- Have one ID and call it the `Request-Id`

Reply:
I tend to agree with this.

Request-ID should be the request-response matching that is generated from the initial `POST` request. Idempotency key will be unique for each leg

Reply:
> Request-ID should be the request-response matching that is generated from the initial POST request

If they're not randomly generated, what if two peers send incoming packets with colliding request IDs that a node routes to the same peer?

Should the `Request-Id` be a u32 to stay consistent across BTP hops as well?

Reply:
Like I mentioned below. Lets just be normative here and say it must be a UUID v4. Is there any reason we don't just make that the requirement?

Reply:
Making UUIDv4 normative sounds fine. Separately, are you saying the request ID should remain the same across hops? If so, how should collisions be handled?

Reply:
Nope request-id is local between two nodes. It is used for the async request-response matching of a single leg and would be the SAME for a single request-response. The Idempotency key then would be unique for the request AND response.

Reply:
Cool, gotcha 👍

Reply:
Separated into two headers and prescribe UUIDv4: https://github.com/interledger/rfcs/compare/060732f4f52f286c020c9ebf085b453e91e51951...f4fc0020d82b46b6fdfde537a8f251906d95120c#diff-ccfda9b7df50a9cfd8618bb9c08e360aR47-R48